torchx/tracker/api.py

```diff
@@ -181,26 +183,28 @@ def build_trackers(
 ) -> Iterable[TrackerBase]:
     trackers = []
 
-    entrypoint_factories = load_group("torchx.tracker")
+    entrypoint_factories = load_group("torchx.tracker") or {}
     if not entrypoint_factories:
-        logger.warning(
-            "No 'torchx.tracker' entry_points are defined. Tracking will not capture any data."
-        )
-        return trackers
+        logger.warning("No 'torchx.tracker' entry_points are defined.")
 
     for entrypoint_key, config in entrypoint_and_config.items():
-        if entrypoint_key not in entrypoint_factories:
-            logger.warning(
-                f"Could not find `{entrypoint_key}` tracker entrypoint. Skipping..."
+        factory = entrypoint_factories.get(entrypoint_key)
+        if factory is None:
+            logger.info(
+                f"Could not find `{entrypoint_key}` among entry_points, checking modules..."
             )
-            continue
-        factory = entrypoint_factories[entrypoint_key]
+            try:
+                factory = hydra.utils.get_method(entrypoint_key)
```

Comment:
could we not depend on `hydra`?

Reply:
Sure, I didn't notice it's in dev-requirements only. So I take no objections about adding this functionality overall, right @kiukchung ?

Reply:
If not and we don't want to stick to Hydra, we have options for the syntax, e.g. 1) `my_tracker = torchx.tracker.mlflow.create_tracker` (Hydra-ish), 2) `my_tracker = torchx.tracker.mlflow:create_tracker` (entrypoint-ish).
Any preference, @kiukchung?